src/explorer.api/Exploration/ExplorationRegistry.cs

```diff
@@ -0,0 +1,116 @@
+namespace Explorer.Api
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Explorer;
+
+    public class ExplorationRegistry
+    {
+        private readonly Dictionary<Guid, Registration> registrations = new Dictionary<Guid, Registration>();
```

Comment:
Good point, I think you're right. I'm not 100% sure but I think that since Asp.Net can process api requests in parallel there's a chance of concurrent access to this variable. I'll change it.

Reply:
9ea163986c5e14d2883190a4216ef8b875cd9992